src/platform_impl/linux/wayland/window/shim.rs

```diff
@@ -188,6 +211,48 @@ impl WindowHandle {
         }
     }
 
+    pub fn set_user_attention(&self, request_type: Option<UserAttentionType>) {
+        let xdg_activation = match self.xdg_activation.as_ref() {
+            None => return,
+            Some(xdg_activation) => xdg_activation,
+        };
+
+        match request_type {
+            None => {
+                if let Some(token) = self.xdg_activation_token.take() {
+                    token.destroy();
+                }
+            }
+            Some(_) => {
+                // The token is already present, nothing to set.
+                if self.xdg_activation_token.borrow().is_some() {
+                    return;
+                }
+
+                let xdg_activation_token = xdg_activation.get_activation_token();
+                let surface = self.window.surface();
+                let window_id = wayland::make_wid(surface);
+                let xdg_activation = xdg_activation.clone();
+                xdg_activation_token.quick_assign(move |xdg_token, event, mut dispatch_data| {
+                    let winit_state = dispatch_data.get::<WinitState>().unwrap();
+                    if let Some(window_handle) = winit_state.window_map.get_mut(&window_id) {
+                        let surface = window_handle.window.surface();
+                        if let xdg_activation_token_v1::Event::Done { token } = event {
```

Comment:
Wouldn't it be better to check for this first, before doing any of the surface acquisition logic? So just a quick `event != Done => return`?